### PR TITLE
[SPARK-23826] [TEST] TestHiveSparkSession should set default session

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
In TestHive, the base spark session does this in getOrCreate(), we emulate that behavior for tests.

## How was this patch tested?
N/A

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @marmbrus @ericl @jose-torres @cloud-fan

**Comment by jose-torres** [comment]
lgtm

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/test/TestHive.scala [hunk 1]
```diff
@@ -159,9 +159,10 @@ private[hive] class TestHiveSparkSession(
     private val loadTestTables: Boolean)
   extends SparkSession(sc) with Logging { self =>
 
-  // TODO(SPARK-23826): TestHiveSparkSession should set default session the same way as
-  // TestSparkSession, but doing this the same way breaks many tests in the package. We need
-  // to investigate and find a different strategy.
+  // The base spark session does this in getOrCreate(), here we emulate that behavior for tests.
+  if (SparkSession.getDefaultSession.isEmpty) {
+    SparkSession.setDefaultSession(this)
+  }
```
Comment (gatorsmile) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/test/TestHive.scala]: This is not needed after we merge https://github.com/apache/spark/pull/20927

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**